### ISSUE TITLE
How to test one by one video?

### ISSUE THREAD

**Issue by JasOlean** [yysijie/st-gcn#195]
### Other comments
After obtaining weights file, how to test one by one video file?

**Comment by daylightsaving2017** [comment]
Due to unknown tricks, I cannot test the video using the **for** loop (because it would always return the inference results of the first video). Hence, I solve this problem by calling the command line inside the python script with different input arguments iteratively.

**Comment by daylightsaving2017** [comment]
@JasOlean

**Comment by JasOlean** [comment]
Where can I reference it?

**Comment by daylightsaving2017** [comment]
@JasOlean Hahhh. Meet you again. I suggest that you could first use openpose to extract the skeletons from all your test videos with storing the result in the 'data/openpose_estimation' directory and then use command line function imported from os package installed in the python environment. About how to implement the virtual input, you could use some string concat functions. 
This idea is also inspired by the author of this project in the openpose calling mechanism which has been provided with the source code.

**Comment by JasOlean** [comment]
Yes. I already done all that steps. I even trained my dataset. But, accuracy is about over 41 %. 
That is why, I want to know what I need to do to obtain better results.

**Comment by JasOlean** [comment]
When I test one by one video, I got memory dump error.

**Comment by daylightsaving2017** [comment]
@JasOlean About the accuracy problem, I advise you to read the corresponding paper more carefully. In the paper, the author wrote a section to explain the low accuray in the object/background-related video. Hence,  this is not the fault or shortcome of the algorithm but the problem of inappropriate application scenes. If you wanna get significantly better recognition results on the current mainstream datasets, you should focus on the CNN(even with optical flow) methods like TSN/ C3D algorithms.

**Comment by daylightsaving2017** [comment]
@JasOlean  About the memory dump error, I am not sure what really invoked the problems by such limited information. However, I suggest you use a GPU with not lower than 8GB display memory.

**Comment by JasOlean** [comment]
Thank you so much for your advice. I will find it.
so, for testing, how to test particular video after getting weights like your demo video. [https://github.com/yysijie/st-gcn/blob/master/resource/info/demo_video.gif]

**Comment by yysijie** [comment]
@JasOlean, @daylightsaving2017,
The latest Openpose provides Python-API. It let the end-to-end inference from video become possible, which may help you in your application.   
See the new demo script mentioned in the README.MD.

**Comment by starry-night-0** [comment]
Hi,
 I have tested  the Openpose Python-API with st-gcn, pretty well, but now I want to use the openpose of tensorflow version, tf-pose-estimation, and  I don't know how to do it on openpose of tensorflow version? How to save the skeletons results in the 'data/openpose_estimation' directory ?
Thanks a lot.
